### PR TITLE
HACK week: use new PHP 8.0/8.1 functions across codebase and add symfony polyfills to enable this

### PR DESCRIPTION
**Pull request by kalessil**
I'm not advocating merging this PR, rather suggesting to the Platform Team to review if this makes sense at all.

#### Changes proposed in this Pull Request

- add Symfony polyfills, which gracefully add the functions for PHP below the required versions
- migrate codebase to use the new functions

#### Testing instructions

- All CI checks shall pass

-------------------

- [ ] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,12 +13,18 @@
...
       "automattic/jetpack-autoloader": "2.10.10",
       "myclabs/php-enum": "1.7.7",
-      "ext-json": "*",
-      "woocommerce/subscriptions-core": "1.1.0"
+      "woocommerce/subscriptions-core": "1.1.0",
+      "symfony/polyfill-php71": "1.20.0",
+      "symfony/polyfill-php72": "1.23.0",
+      "symfony/polyfill-php73": "1.23.0",
+      "symfony/polyfill-php74": "1.23.0",
+      "symfony/polyfill-php80": "1.23.1",
```
Comment (kalessil) [inline composer.json:26]: In our current codebase state, polyfills for PHP 7.1 - 7.4 are not used, so we could drop those.

**Review by vbelolapotkov — APPROVED** [review]
@kalessil the change looks good, but I hesitate releasing it on Dec 29 due to limited team coverage if we missed something. WDYT about releasing it in Jan?

Pre-approving though.

File: composer.json (hunk 1 above)
Comment (vbelolapotkov) [reply composer.json:26]: I haven't double checked this, but trust you 🙂

**Comment by kalessil** [comment]
:+1: For January, thanks for reviewing!

**Pull request merged**